### PR TITLE
Add equations by Lucas

### PR DESCRIPTION
**Pull request by benegee**
I propose to add the moist Euler equations implemented by Lucas, as-is, as a first step. @Arpit-Babbar and @FnHck then have a common basis for their experiments.

Note that we should review, clean up, ... this at some point. But maybe we can postpone this until we start implementing alternative equations.

### CODE REVIEW

**Comment by benegee** [comment]
# Dry air consistency check

I also added a first test, which runs Trixi.jl's `elixir_euler_warm_bubble.jl` and then runs the same elixir again using the equations defined in TrixiAtmo.jl. Thereby we can make sure that the special case of a dry atmosphere, with the additional components set to 0, yields the same results a the ordinary Euler equations in Trixi.jl.

`trixi_include` comes in handy here as it allows to overwrite the elixir an the fly. However, this can certainly be improved. See e.g. https://github.com/trixi-framework/TrixiAtmo.jl/actions/runs/10007442043/job/27662157328?pr=21#step:5:690

File: src/equations/compressible_moist_euler_2d_lucas.jl [hunk 1]
```diff
@@ -0,0 +1,1032 @@
...
+        @unpack g = equations
+        rho, rho_v1, rho_v2, rho_e, rho_qv, rho_ql = u
+        tmp = rho_v2
+
+        return SVector(zero(eltype(u)), zero(eltype(u)),
+                       -g * rho, -g * tmp,
+                       zero(eltype(u)), zero(eltype(u)))
+    end
+
+    # Raylight damping sponge source term form A. Sridhar et al.,
```
Comment (tristanmontoya) [inline src/equations/compressible_moist_euler_2d_lucas.jl:356]:
```suggestion
    # Rayleigh damping sponge source term form A. Sridhar et al.,
```

File: src/equations/compressible_moist_euler_2d_lucas.jl [hunk 2]
```diff
@@ -0,0 +1,1032 @@
...
+    # Large-eddy simulations with ClimateMachine: a new open-sourcecode for
+    # atmospheric simulations on GPUs and CPUs, 2 Oct 2021, doi: 10.5194/gmd-15-6259-2022,
+    # https://arxiv.org/abs/2110.00853 [physics.ao-ph] .
+    @inline function source_terms_nonhydrostatic_raylight_sponge(u, x, t,
```
Comment (tristanmontoya) [inline src/equations/compressible_moist_euler_2d_lucas.jl:360]:
```suggestion
    @inline function source_terms_nonhydrostatic_rayleigh_sponge(u, x, t,
```

**Review by tristanmontoya — CHANGES_REQUESTED** [review]

File: test/test_trixi_consistency.jl
[the remark this answers is not in the document]
Comment (benegee) [reply test/test_trixi_consistency.jl:1]: Sorry for that! In a way this PR was not yet ready to go as I was not very confident about the double trixi_include.

The errors were due to a different parameter setting. This is now adapted and allows to further decrease the tolerances. Maybe we should think about how strict the tolerances should be. We could of course just adopt the Trixi.jl defaults.

**Review by tristanmontoya — APPROVED** [review]

**Review by tristanmontoya — APPROVED** [review]

**Review by tristanmontoya — APPROVED** [review]

**Pull request merged**